### DISCUSSION TITLE
Collecting definitions / references defined by an `exposer`

### DISCUSSION THREAD

**Discussion by moxley01** [swaggest/openapi-go#113] [Q&A]
Hello,

Say we have a struct that declares its own schema by virtue of being an `exposer` (i.e. implements `JSONSchema() (jsonschema.Schema, error) `). And suppose that this json-schema in turn relies on its own set of definitions / references, how do we ensure that those definitions end up in the final schema file ? Right now I am finding that they get stripped, leaving an invalid spec with unresolved references.

**Comment by vearutop** [comment]
This is complicated to automate, because references in JSON schema are usually placed differently than in OpenAPI (not in `#/components/schemas`).

Could you share an example of such an exposer to have a better understanding of the case?
